src/zambeze/cli.py

```diff
@@ -47,11 +47,15 @@ def start():
         logger.error(f"Failed to create log directory at {logs_base_dir}")
         return
 
-    # Create directory for log files
+    # Create path for log file
     fmt = datetime.now().strftime("%Y_%m_%d-%H_%M_%S_%f")[:-3]
-    log_dir_path = pathlib.Path.home() / ".zambeze/logs" / fmt
-    log_dir_path.mkdir(exist_ok=True)
-    log_path = log_dir_path / "zambeze.log"
+    log_path = pathlib.Path.home() / ".zambeze/logs" / fmt / "zambeze.log"
+    try:
+        log_path.parent.mkdir(exist_ok=True)
+        log_path.touch(exist_ok=True)
+    except OSError:
+        logger.error(f"Failed to create log file at {log_path}")
+        return
```

Comment:
> The mkdir is explicity raising OsError; FileNotFoundError won't raise in our case because we have taken care of creating a logs_base_dir before creating this directory. The touch is using os.O_CREAT and os.O_WRONLY flags. While digging up further, I found that the flags could open different possible exceptions subclassed under OsError (eg. PermissionError, NotADirectoryError, ENOSPC (No space), etc.). I am thinking it's probably a good idea to catch the general OsError, please let me know your thoughts.

This is from @flourishingtune comment below. Putting here because it's related to this conversation.